cli/file_fetcher.rs

```diff
@@ -383,44 +383,74 @@ impl FileFetcher {
     async move {
-      let result = match fetch_once(
-        &client,
-        FetchOnceArgs {
-          url: specifier.clone(),
-          maybe_accept: maybe_accept.clone(),
-          maybe_etag,
-          maybe_auth_token,
-          maybe_progress_guard: maybe_progress_guard.as_ref(),
-        },
-      )
-      .await?
-      {
-        FetchOnceResult::NotModified => {
-          let file = file_fetcher.fetch_cached(&specifier, 10)?.unwrap();
-          Ok(file)
-        }
-        FetchOnceResult::Redirect(redirect_url, headers) => {
-          file_fetcher.http_cache.set(&specifier, headers, &[])?;
-          file_fetcher
-            .fetch_remote(
-              &redirect_url,
-              permissions,
-              redirect_limit - 1,
-              maybe_accept,
-            )
-            .await
-        }
-        FetchOnceResult::Code(bytes, headers) => {
-          file_fetcher
-            .http_cache
-            .set(&specifier, headers.clone(), &bytes)?;
-          let file =
-            file_fetcher.build_remote_file(&specifier, bytes, &headers)?;
-          Ok(file)
-        }
+      let mut retried = false;
+      let result = loop {
+        let result = match fetch_once(
+          &client,
+          FetchOnceArgs {
+            url: specifier.clone(),
+            maybe_accept: maybe_accept.clone(),
+            maybe_etag: maybe_etag.clone(),
+            maybe_auth_token: maybe_auth_token.clone(),
+            maybe_progress_guard: maybe_progress_guard.as_ref(),
+          },
+        )
+        .await?
+        {
+          FetchOnceResult::NotModified => {
+            let file = file_fetcher.fetch_cached(&specifier, 10)?.unwrap();
+            Ok(file)
+          }
+          FetchOnceResult::Redirect(redirect_url, headers) => {
+            file_fetcher.http_cache.set(&specifier, headers, &[])?;
+            file_fetcher
+              .fetch_remote(
+                &redirect_url,
+                permissions,
+                redirect_limit - 1,
+                maybe_accept,
+              )
+              .await
+          }
+          FetchOnceResult::Code(bytes, headers) => {
+            file_fetcher
+              .http_cache
+              .set(&specifier, headers.clone(), &bytes)?;
+            let file =
+              file_fetcher.build_remote_file(&specifier, bytes, &headers)?;
+            Ok(file)
+          }
+          FetchOnceResult::RequestError(err) => {
+            // Retry once, and bail otherwise.
+            if !retried {
+              retried = true;
+              continue;
+            } else {
+              Err(generic_error(format!(
+                "Import '{}' failed: {}",
+                specifier, err
+              )))
+            }
+          }
+          FetchOnceResult::ServerError(status) => {
+            // Retry once, and bail otherwise.
+            if !retried {
+              retried = true;
+              continue;
```

Comment:
Perhaps we should `log::warn` when this occurs?

Reply:
Done

Reply:
Now that I'm thinking about it - a flake will cause this to be printed which will still fail tests. Maybe it should be `log::debug` instead?